src/monodroid/jni/xamarin-app.hh

```diff
@@ -106,6 +108,86 @@ struct XamarinAndroidBundledAssembly final
 	char    *name;
 };
 
+//
+// Blob format
```

Comment:
Format description could also use a "fleshed out" example, e.g. what structs, struct order, and struct values would one expect for `App.dll` & `mscorlib.dll`?  Where do satellite assemblies fit in?  Or do they just "fall out" because their "relative pathname" is hashed, and it's the hash that we care about?

Are there any checks for hash collisions, and an "appropriate error" when a collision is detected?

Reply:
> Format description could also use a "fleshed out" example, e.g. what structs, struct order, and struct values would one expect for `App.dll` & `mscorlib.dll`? Where do satellite assemblies fit in? Or do they just "fall out" because their "relative pathname" is hashed, and it's the hash that we care about?
> 
> Are there any checks for hash collisions, and an "appropriate error" when a collision is detected?

I was focused on implementing the thing first instead of writing and rewriting documents... Once I'm sure that the format is final, I can spend time on documentation

Reply:
> Format description could also use a "fleshed out" example, e.g. what structs, struct order, and struct values would one expect for `App.dll` & `mscorlib.dll`? Where do satellite assemblies fit in? Or do they just "fall out" because their "relative pathname" is hashed, and it's the hash that we care about?
> 
> Are there any checks for hash collisions, and an "appropriate error" when a collision is detected?

Yes there are, [here](https://github.com/xamarin/xamarin-android/pull/6311/files#diff-1b9c459e1ad04c80fa943403a2facf8fb0f726e5b086a004f59d731e3cb80598R124-R134) and [here](https://github.com/xamarin/xamarin-android/pull/6311/files#diff-1b9c459e1ad04c80fa943403a2facf8fb0f726e5b086a004f59d731e3cb80598R168-R177). This code generates the **global** index (contained only in a single blob, the blob with ID 0) spanning all of the blobs that are generated for the application